### PR TITLE
Datalad provider stb

### PR DESCRIPTION
**Pull request by shots47s**
An updated Datalad provider that does all of the necessary things to cache the datalad repo for maintaining the file information inside the datasets.

This is only capable of reading from a datalad dataset now, writing comes next.

### CODE REVIEW

**Comment by shots47s** [comment]
I put it in the main CBRAIN directories as datalad is technically not a neuroimaging tool.

**Comment by prioux** [comment]
Shawn, please return this code to the neuro plugin repo, we are losing all the history of the datalad DP code with this new commit. Also, you have left lots of debug code in here and changed the behavior of the platform (see comments soon).

File: BrainPortal/app/controllers/data_providers_controller.rb [hunk 1]
```diff
@@ -468,10 +469,12 @@ def register
     succeeded, failed = [], {}
 
     CBRAIN.spawn_with_active_records_if(
-      [:html, :js].include?(request.format.to_sym),
+      #[:html, :js].include?(request.format.to_sym),
```
Comment (prioux) [inline BrainPortal/app/controllers/data_providers_controller.rb]: This makes the Rails app block forever

File: BrainPortal/app/controllers/data_providers_controller.rb [hunk 2]
```diff
@@ -468,10 +469,12 @@ def register
...
+      false,
       current_user,
       "Register files DP=#{@provider.id}"
     ) do
+      binding.pry
```
Comment (prioux) [inline BrainPortal/app/controllers/data_providers_controller.rb]: This makes the rails app block forever expecting a programmer to enter the console, which doens't exist in production

File: BrainPortal/app/controllers/data_providers_controller.rb [hunk 3]
```diff
@@ -617,6 +620,7 @@ def unregister
       current_user,
       "Unregister files DP=#{@provider.id}"
     ) do
+      puts "FUCK ME I AM HERE"
```
Comment (prioux) [inline BrainPortal/app/controllers/data_providers_controller.rb]: Bad

**Comment by shots47s** [comment]
This doesn't belong  in cbrain-plugins-neuro, can it be moved after we finalize?

File: BrainPortal/db/schema.rb [hunk 4]
```diff
@@ -96,19 +96,21 @@
     t.string   "remote_host"
     t.integer  "remote_port"
     t.string   "remote_dir"
-    t.boolean  "online",                                        default: false, null: false
-    t.boolean  "read_only",                                     default: false, null: false
-    t.text     "description",                     limit: 65535
+    t.boolean  "online",                                         default: false, null: false
```
Comment (prioux) [inline BrainPortal/db/schema.rb]: Fix so that no chanes are made to the schema when unnecessary

**Pull request closed without merging**